File: patterns/expire-binding-based-timer/rules-python/expire_binding_based_timer.py
from openhab import rule, Registry
from openhab.triggers import ItemCommandTrigger


@rule(triggers=[ItemCommandTrigger("StartMyTimerTrigger", "ON")])
class EineRegelDieDenTimerStartet:
    def execute(self, module, input):
        # Arbeitsschritte ausfuehren

        if str(Registry.getItem("MyTimer").getState()) == "ON":
            self.logger.info("Timer ist bereits aktiv - wird neu gestartet")

        # Timer abbrechen
        Registry.getItem("MyTimer").postUpdate("OFF")

        # Timer starten
        Registry.getItem("MyTimer").sendCommand("ON")
        self.logger.info("MyTimer gestartet (5 Minuten)")


@rule(triggers=[ItemCommandTrigger("MyTimer", "OFF")])
class MyTimerAbgelaufen:
    def execute(self, module, input):
        self.logger.info("MyTimer abgelaufen - Code nach Ablauf wird ausgefuehrt")

File: patterns/expire-binding-based-timer/rules-js/expire-binding-based-timer.js
rules.JSRule({
  name: "Eine Regel, die den Timer startet",
  triggers: [triggers.ItemCommandTrigger("StartMyTimerTrigger", "ON")],
  execute: (event) => {
    // Arbeitsschritte ausfuehren

    if (items.getItem("MyTimer").state === "ON") {
      console.log("Timer ist bereits aktiv - wird neu gestartet");
      // Aktion, falls Timer aktiv ist
    }

    // Timer abbrechen
    items.getItem("MyTimer").postUpdate("OFF");

    // Timer starten
    items.getItem("MyTimer").sendCommand("ON");
    console.log("MyTimer gestartet (5 Minuten)");
  }
});

rules.JSRule({
  name: "MyTimer abgelaufen",
  triggers: [triggers.ItemCommandTrigger("MyTimer", "OFF")],
  execute: (event) => {
    console.log("MyTimer abgelaufen - Code nach Ablauf wird ausgefuehrt");
    // Code, der nach Ablauf ausgefuehrt werden soll
  }
});
